client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -229,9 +229,14 @@ const PatternAssembler = ( {
 		}
 	};
 
+	const activateFooterPosition = ( hasFooter: boolean ) => {
+		const lastPosition = hasFooter ? sections.length : sections.length - 1;
+		updateActivePatternPosition( lastPosition );
+	};
+
 	const updateFooter = ( pattern: Pattern | null ) => {
 		setFooter( pattern );
-		updateActivePatternPosition( sections.length );
+		activateFooterPosition( !! pattern );
```

Comment:
There was a bug that did not scroll to bottom when removing footer. This change fixes it.